### PR TITLE
Log SOF timestamps

### PR DESCRIPTION
**Pull request by shuchitak**
Pass timestamps at which SOFs are received to tud_xcore_sof_cb(). These can be used by the application to calculate the USB rate.

https://xmosjira.atlassian.net/browse/AP-128

### CODE REVIEW

File: test/rtos_drivers/usb/src/individual_tests/usb/local/usb_sof_test.c [hunk 1]
```diff
@@ -38,8 +38,9 @@ static void timeout_cb(TimerHandle_t xTimer)
     sof_timeout = true;
 }
 
-bool tud_xcore_sof_cb(uint8_t rhport)
+bool tud_xcore_sof_cb(uint8_t rhport, uint32_t cur_time)
```
Comment (ACascarino) [inline test/rtos_drivers/usb/src/individual_tests/usb/local/usb_sof_test.c:41]: does it matter that the name of the variable is different between its weak declaration and its definition here?

**Review by ACascarino — COMMENTED** [review]
One comment, happy to approve once cleared up

File: test/rtos_drivers/usb/src/individual_tests/usb/local/usb_sof_test.c (hunk 1 above)
Comment (ACascarino) [reply test/rtos_drivers/usb/src/individual_tests/usb/local/usb_sof_test.c:41]: Actually looking at the other review, you've used "cur_time" as the variable name everywhere else - suggest changing the declaration to "cur_time" to harmonise throughout

Comment (shuchitak) [reply test/rtos_drivers/usb/src/individual_tests/usb/local/usb_sof_test.c:41]: I'll change it. About your earlier comment I don't think the variable names differently in the weak declaration matters.

**Review by ACascarino — APPROVED** [review]

**Pull request merged**